lms/djangoapps/instructor/views/api.py

```diff
@@ -3592,3 +3596,151 @@ def _create_error_response(request, msg):
     in JSON form.
     """
     return JsonResponse({"error": _(msg)}, 400)
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('staff')
+@require_http_methods(['POST'])
+@require_post_params(action="ban or unban", identifiers="stringified list of emails and/or usernames and expiry time")
+def ban_student_from_course(request, course_id):
+    """
+    Ban the student from the particular course.
+    :param request: HttpRequest object,
+    :param course_id: course identifier for the course for whom to ban the student,
+    :return: JsonResponse object containing the success/faliure message and exception data
+    """
+    if request.method == 'POST':
+        allowed_actions = ['ban', 'unban']
+        action = request.POST.get('action')
+
+        if action not in allowed_actions:
+            return JsonResponse({'message': 'Bad Action type'}, status=422)
+
+        successful_usernames, invalid_usernames = [], []
+        course_id = CourseKey.from_string(course_id)
+        usernames_or_expires = _clean_username_and_expiry(request.POST.get('identifiers'))
+
+        if action == 'ban':
+            successful_usernames, invalid_usernames = _ban_username_or_email(course_id, usernames_or_expires)
+
+        elif action == 'unban':
+            successful_usernames, invalid_usernames = _unban_email(usernames_or_expires)
+
+        results = {
+            'action': action,
+            'failed_results': invalid_usernames,
+            'successful_results': successful_usernames
+        }
+
+        return JsonResponse(results)
```

Comment:
I guess passing the `status_code` will be a better approach. We might need it in the future.

Reply:
it will get the status code `200` through django itself